src/e4c_pk_store_mem.c

```diff
@@ -299,16 +304,39 @@ int e4c_reset_devices(e4storage* store)
 }
 
 int e4c_set_c2_pubkey(e4storage* store, const uint8_t* key) {
+    size_t devicekeynotempty = 0;
     memcpy(store->c2key, key, E4_PK_X25519_PUBKEY_LEN);
     e4c_sync(store);
-    return 0;
+    devicekeynotempty = zerocheck(store->privkey, sizeof(store->privkey));
```

Comment:
Same comment as above, more or less. This is the inverse case: the c2 public key is being set, but we don't know if the device privkey has been set yet (maybe not). So we check. If it is all zero, we don't do any derivation.

Note that if you do e4c_load() from the backing storage, this also computes the shared key because it is assumed that if the data has been written to the file, then at that point it is probably OK to do so. I should probably add zero-checks there as well.

Reply:
Where we should error, actually, is if the shared point is all zero. I'll add this.

Reply:
This wasn't necessary, we already error if the shared point is zero. What we did need to check was the load issue, now done.